src/service.rs

```diff
@@ -309,36 +309,39 @@ where
     >>::Error: Display,
     for<'a> <TYPES::SignatureKey as TryFrom<&'a TaggedBase64>>::Error: Display,
 {
-    async fn bundle(&self, view_number: u64) -> Result<Bundle<TYPES>, BuildError> {
+    #[tracing::instrument(skip(self))]
+    async fn bundle(
+        &self,
+        parent_view: u64,
+        parent_hash: &VidCommitment,
+        _view_number: u64,
+    ) -> Result<Bundle<TYPES>, BuildError> {
         let start = Instant::now();
-        let requested_view = TYPES::Time::new(view_number);
-        // TODO: We should either change the API to match leader's intent or track the builder states
-        // differently, as this won't work for a view immediately after a view failure.
-        let parent_view = TYPES::Time::new(view_number.saturating_sub(1));
+
+        let parent_view = TYPES::Time::new(parent_view);
+        let state_id = BuilderStateId {
+            view: parent_view,
+            parent_commitment: parent_hash.clone(),
+        };
```

Comment:
I found these field names a bit confusing so wanted to double-check. With `view` and `parent_commitment`, I would assume `view` is the current view number, i.e., it's newer than the view corresponding to `parent_commitment`. However, here it looks like they both correspond to the same parent.

If it's just a naming issue, I think it's fine to leave it as is, and maybe update the field name in a separate PR. But I'm wondering if the `parent_commitment` should actually be the parent of the parent, i.e., the grandparent commitment.

Reply:
Yeah, you're right, this is unfortunate naming. I'd vote for renaming view to parent_view for minimal confusion, wdyt?

Reply:
Maybe rename `parent_commitment` to `commitment`? So that we can look up builder info by `BuilderStateId` for any view, not necessarily the parent view. If we only do lookups for the parent view, then yes, renaming `view` to `parent_view` would be better.

Reply:
Yeah, we always look up for parent view, BuilderState is identified by view and commitment of the parent block. Renamed.